### PR TITLE
fix: `GLOBUS_SDK_ENVIRONMENT` environment variable will now take precedence over `GLOBUS_SDK_OPTIONS`

### PR DESCRIPTION
**Pull request by jakeglobus**
- precedence should have the ENV var above config object

### CODE REVIEW

File: src/lib/core/__tests__/global.spec.ts [hunk 1]
```diff
@@ -34,6 +34,22 @@ describe('getEnvironment', () => {
       getEnvironment();
     }).toThrow(EnvironmentConfigurationError);
   });
+
+  it('should take precedence over SDKOptions object', () => {
```
Comment (jbottigliero) [inline src/lib/core/__tests__/global.spec.ts:38]:
```suggestion
  it('uses GLOBUS_SDK_ENVIRONMENT over GLOBUS_SDK_OPTIONS for sourcing environment', () => {
```

File: src/lib/core/__tests__/global.spec.ts [hunk 2]
```diff
@@ -34,6 +34,22 @@ describe('getEnvironment', () => {
...
+    process.env['GLOBUS_SDK_OPTIONS'] = JSON.stringify({
+      environment: ENVIRONMENTS.PREVIEW,
+    });
+    process.env['GLOBUS_SDK_ENVIRONMENT'] = ENVIRONMENTS.SANDBOX;
+    expect(getEnvironment()).toEqual(ENVIRONMENTS.SANDBOX);
+  });
+
+  it('should be sourced from SDKOptions object when GLOBUS_SDK_ENVIRONMENT var is not set', () => {
```
Comment (jbottigliero) [inline src/lib/core/__tests__/global.spec.ts:46]:
```suggestion
  it('should be sourced from GLOBUS_SDK_OPTIONS when GLOBUS_SDK_ENVIRONMENT is not set', () => {
```

**Review by jbottigliero — APPROVED** [review]
Minor language tweaks to clarify the test behaviors.

**Comment by jakeglobus** [comment]
What do you think about a `console.warn` in `getEnvironment()` that lets them know that both values are set and are set to different value?

**Review by theroncross — APPROVED** [review]

**Comment by jbottigliero** [comment]
> What do you think about a `console.warn` in `getEnvironment()` that lets them know that both values are set and are set to different value?

You could do: `import { log } from '../logger.js'; log('debug', ...);`

**Pull request merged**